### ISSUE TITLE
levelDirector.loadLevel needs to be deferred

### ISSUE THREAD

**Issue by parasyte** [melonjs/melonJS#455]
See also #396.

`me.levelDirector.loadLevel` is almost always called by some object within the game loop update cycle. This is bad because the call stack looks like this (taken during plaformer demo's level transition) :

```
me.levelDirector.obj.loadLevel (melonJS-1.0.0.js:15284)
me.LevelEntity.me.ObjectEntity.extend.onFadeComplete (melonJS-1.0.0.js:7601)
update (melonJS-1.0.0.js:15805)
me.ObjectContainer.me.Renderable.extend.update (melonJS-1.0.0.js:6608)
me.game.api.update (melonJS-1.0.0.js:1275)
_renderFrame (melonJS-1.0.0.js:7829)
```

This means `me.game.update` method is currently iterating all objects. `me.levelDirector.loadLevel` calls `me.game.reset` in the middle of the loop. When the loadLevel method returns up the stack to the game loop (named `me.ObjectContainer.me.Renderable.extend.update` in the stack trace), it will continue the iteration using *brand new* objects, starting from an arbitrary index. This is undefined behavior.

To be fair, `me.levelDirector.loadLevel` already stops the game loop. However, this won't matter because loadLevel is not asynchronous; it runs from start to end without waiting on any events. That makes it impossible for the next animation frame callback to start running while a level is being loaded.

Recommended solution: Use a better event manager (#396) to schedule loadLevel internally, such that it can freely wipe out all game objects without breaking iteration. Also remove the state stop/restart.
